docs/docs/configuration.md

```diff
@@ -164,6 +172,40 @@ export default {
 } satisfies SkuConfig;
 ```
 
+## dangerouslySetViteConfig
+
+Type: `function`
+
+Bundler: `vite`
+
+This function provides a way to modify sku's Vite configuration.
+It should only be used in exceptional circumstances where a solution cannot be achieved by adjusting standard configuration options.
+
+Before customizing your Vite configuration, please reach out in [#sku-support](https://seek.enterprise.slack.com/archives/CDL5VP5NU) to discuss your requirements and potential alternative solutions.
+
+As sku creates two Vite configs (`client` & `render`), this function will actually run twice.
+If you only need to modify one of these configs, then you can check `env.mode` from the second argument within.
+
+This function can return a partial config object that will be deeply merged into existing config (recommended), or directly mutate the config (if the default merging cannot achieve the desired result).
+
+> Sku provides no guarantees that its Vite configuration will remain compatible with any customizations made within this function.
+> It is the responsibility of the user to ensure that their customizations are compatible with sku.
+
+Example:
+
+```ts
+export default {
+  // partial config is deeply merged
+  dangerouslySetViteConfig: () => ({
```

Comment:
It may be beneficial to list what args are provided here


```suggestion
  dangerouslySetViteConfig: (config, env) => ({ // or whatever the actual args are
```